idelib/util.py

```diff
@@ -35,80 +37,154 @@ def verify(data, schema=None):
+
+                if n % increment == 0:
+                    updater(count=n, total=totalElements)
+
+            if el.name == 'ChannelDataBlock':
+                # Get ChannelIDRef, StartTimeCodeAbs, and EndTimeCodeAbs;
+                # usually the 1st three, in order, but don't assume!
+                chId = blockStart = blockEnd = None
+                for subEl in el:
+                    if subEl.name == "ChannelIDRef":
+                        chId = subEl.value
+                    elif subEl.name == "StartTimeCodeAbs":
+                        blockStart = subEl.value
+                    elif subEl.name == "EndTimeCodeAbs":
+                        blockEnd = subEl.value
+
+                blockEnd = blockEnd or blockStart
+                if chId is None:
+                    logger.warning("Extractor: {} missing <ChannelIDRef> subelement, skipping.".format(el))
+                    continue
+                if blockStart is None:
+                    logger.warning("Extractor: {} missing <StartTimeCodeAbs> subelement, skipping.".format(el))
+                    continue
+
+                if channels and chId not in channels:
+                    continue
+                if finished.setdefault(chId, False):
+                    continue
+
+                # TODO: Modulus correction, if still needed.
+                scalar = timeScalars.get(chId, 1)
+                blockStart *= scalar
+                blockEnd *= scalar
+
+                writeCurrent = True
+                writePrev = False  # write previous block, if current one starts late
+
+                if blockEnd < startTime:
+                    # Entirely before extraction interval. Write nothing.
+                    writeCurrent = False
+                elif blockStart <= startTime:
+                    # Block overlaps start of interval. Write block.
+                    # Mark channel finished if block also includes end of interval.
+                    writePrev = False
+                    finished[chId] = blockEnd >= endTime
+                elif blockEnd <= endTime:
+                    # Block within interval. Write block (w/ prev. if needed).
+                    writePrev = channelsWritten[chId] == 0
+                else:
+                    # Block overlaps end of interval. Write block (w/ prev. if needed).
+                    # Mark channel finished.
+                    finished[chId] = True
+                    writePrev = channelsWritten[chId] == 0
+
+                if writePrev:
+                    # Write the previous block, which is before the extraction interval.
+                    # This is to ensure that initial data is not left out.
+                    prev = lastBlocks.get(chId, None)
+                    if prev:
+                        data = prev[0].getRaw()
+                        fs.write(data)
+                        copiedBytes += len(data)
+                        channelsWritten[chId] += 1
```

Comment:
I'm not sure I understand why we want to do this.  I would expect to have no data on a channel if it has no data within the specified range, rather than some data before the cutoff.

Reply:
Primarily, this is for the temperature-compensating bivariate polynomials. Also, I believe users will expect some data, even if the channel doesn't have samples that happen to fall within the specified interval. There have been some complaints when they 'lose' data from the start or end of a split recording.

Reply:
The bivariates definitely make sense.  It looks like if all the blocks on a channel end before the start of the interval you're extracting, it could end up writing none anyway.

Reply:
I still think that this could cause an issue if someone wants to export acceleration data that relies on the internal temperature, but only exists after the temperature sensor has no data.

Reply:
Unless the file is damaged, there will always be temperature data before the analog accelerometer channel that needs it. Even if the last temperature sample came before the extracted range, the file will contain that last block. If the method `EventArray.getEventIndexNear()` (used by the bivariate polynomial) is called with a time beyond the actual range, it will return either the first or last.

tl;dr: It shouldn't be an issue, and if it is, it's a bug elsewhere in `idelib`.